worker/codalabworker/aws_batch.py

```diff
@@ -368,11 +374,14 @@ def get_job_definition(self):
             name=self.uuid,
             read_only=False
         )]
-        for host_path, docker_path, uuid in dependencies:
+        for host_path, docker_path, name in dependencies:
+            # Batch has some restrictions with the name, so force it to conform
+            # See: https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#volumes
+            clean_name = re.sub(r'[^a-zA-Z0-9_-]', '', name)[:244]
             volumes_and_mounts.append(self.volume_and_mount(
                 host_path=host_path,
                 container_path=docker_path,
-                name='dependency_'+uuid,
+                name='dependency_'+clean_name,
```

Comment:
Why was this a problem before, since `uuid`s should already be clean, no?

Reply:
They weren't actually uuids, the name was misleading. This was actual the name alias. So in a command like `cl run foo:0x12345 ..` it would be `foo`.

Reply:
Does this name need to be unique? If so we maybe should add a loop index at the end or something?